Servers/WindowServer/WSWindowManager.cpp

```diff
@@ -489,10 +489,14 @@ void WSWindowManager::notify_occlusion_state_changed(WSWindow& window)
 
 void WSWindowManager::pick_new_active_window()
 {
+    bool new_window_picked = false;
     for_each_visible_window_of_type_from_front_to_back(WSWindowType::Normal, [&](WSWindow& candidate) {
         set_active_window(&candidate);
+        new_window_picked = true;
         return IterationDecision::Break;
     });
+    if (!new_window_picked)
+        set_active_window(nullptr);
```

Comment:
This is a good fix. There's a hack in `remove_window()` that works around the fact that this wasn't happening.
So please remove that hack when fixing this :)